<launch>
  <arg name="input_cmd" value="1" /> <!-- Set input mode (0 for velocity/steer, 1 for force/steer) -->

  <!--- Run single track simulator node -->
  <node pkg="singletrack_sim" type="singletrack_beta_sim" respawn="false" name="singletrack_beta_sim" output="screen">
    <rosparam file="$(find singletrack_sim)/config/singletrack_beta_sim.yaml" command="load" />
    <rosparam param="/use_sim_time">true</rosparam>
    <rosparam param="/singletrack_beta_sim/input_cmd">1</rosparam> <!-- Set input mode to force -->
    <rosparam param="/singletrack_beta_sim/steering_actuator_model">0</rosparam> <!-- Set ideal steering actuator -->
    <rosparam param="/singletrack_beta_sim/tyre_model">0</rosparam> <!-- Set linear tyre model -->
    <rosparam param="/singletrack_beta_sim/automode_delay">0.0</rosparam> <!-- Set delay before starting autonomous mode -->
    <rosparam param="/singletrack_beta_sim/V0">1.0</rosparam> <!-- Set initial longitudinal velocity -->
  </node> 

  <!--- Run test single track simulator node -->
  <node pkg="singletrack_sim" type="test_singletrack_sim" respawn="false" name="test_singletrack_sim" output="screen">
    <rosparam param="/test_singletrack_sim/input_cmd">1</rosparam> <!-- Set input mode to force -->
  </node> 
</launch>
